packages/echo-core/src/types/registry/contextualAppLink.types.ts

```diff
@@ -0,0 +1,6 @@
+export type ContextualAppLinkExtensionOptions = {
+    iconName?: string;
+    label?: string;
+    appPath?: string;
+    isExternalLink?: boolean;
```

Comment:
appPath and isExternalLink, they are not required?
Optional sounds a bit strange, but maybe I don't see the full picture

Reply:
In case of `isExternalLink`: if this is not provided, then the link will be treated as internal link. That's why it's optional.
`appPath`: if a contextual app link extension contains a component, then in this case it's expected that the passed component handles the navigation with app path included.
In this scenario `appPath` is not needed: so it's optional.